### ISSUE TITLE
Synchronized #include ignore file does not trigger update

### ISSUE THREAD

**Issue by skst** [syncthing/syncthing#7096]
(This is my first bug report on syncthing. I apologize if I am not using the correct terms. I am happy to correct the report and, of course, to add any information I have neglected to include.)

### Versions

Syncthing x64 **1.11.1**
SyncTrayzor x64 **1.1.23**
Computer A: Windows 8.1 Pro with Pro Pack
Computer B: Windows 10 Home x64 20H2 (19042.572)
Computer C: Windows 10 Pro x64 20H2 (19042.572)

### Summary

The goal is to use Computer C to control files synchronized from Computer A to Computer B.

Updating the #included stignore.txt on Computer C should trigger Computer B to pull files from Computer A.

### Details

**Syncthing A**
Send-only: Computer A can send any file to Computers B and C.

**Syncthing B**
Send & Receive: Computer B receives only certain files from Computer A and synchronizes _stignore.txt_ with Computer C.

_Ignore Patterns:_
```
#include stignore.txt
*.wtv
```

_stignore.txt_
The ignore file allows specific _.wtv_ files to be synchronized from Computer A to Computer B.
```
!(?i)Some File*.wtv
*.wtv
```

**Syncthing C**
Send & Receive: Computer C synchronizes only _stignore.txt_ with Computer B (but _could_ receive files from Computer A).

_Ignore Patterns:_
```
*.wtv
```

The _stignore.txt_ is identical to the one on Computer B.

**Steps**
Use Computer C to edit the _stignore.txt_ file and add another file to be synchronized from Computer A to Computer B.

_stignore.txt_
```
!(?i)Some File*.wtv
!(?i)New File*.wtv
*.wtv
```

**Results**
The stignore.txt file is synchronized from Computer C to Computer B.
**Bug:** Computer B ignores the updated _stignore.txt_ file and does _not_ pull "New File.wtv" from Computer A.
I must manually tell Syncthing to rescan the folder in order to start the synchronization.

**Desired Behavior**
Syncthing should detect that an #included file has changed and automatically rescan the folder.

This feature is specifically called out in the documentation for _Ignoring Files_:
> The .stignore file itself will never be synced to other devices, although it can #include files that are synchronized between devices.

**Comment by calmh** [comment]
The ignore file is re-read only on rescans, which happen periodically or when a (non-ignore) file changes.

I suppose that in the case where we have fs notifications we could also monitor the ignore file and have that trigger a full rescan.

**Comment by skst** [comment]
> The ignore file is re-read only on rescans, which happen periodically or when a (non-ignore) file changes. Whether it or its includes are synced doesn't factor into this.

To be clear, I'm synchronizing the _stignore.txt_ file which is #included by _.stignore_. I don't expect the _.stignore_ file itself to be synchronized or re-read. However, when a file that the _.stignore_ file #includes (_e.g., stignore.txt_) changes (due to synchronization), my expectation is that the _.stignore_ file would be re-read because a non-ignore file changed. If I understand you correctly, you're saying that should happen, but it is not working in my scenario.

**Comment by calmh** [comment]
No, I'm saying ignores are re-read when a scan happens, but changing the ignore file (or it's includes) doesn't necessary trigger a scan.

**Comment by imsodin** [comment]
When syncing/pulling the included file, we do detect that ignores changed. However as calmh writes, that does not currently trigger a scan. And a scan is needed to "unignore" the now excluded file, so that it gets synced.  
Similar thing when one edits an included file: With watching for changes a scan of that file is triggered, in which we detect that the ignore patterns changed, but again we don't issue a full scan because of that.    
When changing ignores in the UI/through the API, we do trigger a full scan.  

In my opinion it would be a good idea to also do a full scan if we detect changed ignores during pulling and during partial scans.
